courses/sync_external_courses/emeritus_api.py

```diff
@@ -400,20 +403,25 @@ def create_or_update_emeritus_course_page(course_index_page, course, emeritus_co
         )
         created = True
     else:
-        # Only update course page fields with API if they are empty.
+        latest_revision = course_page.get_latest_revision_as_object()
+        # Only update course page fields with API if they are empty in the latest revision.
         course_page_attrs_changed = False
-        if not course_page.external_marketing_url and emeritus_course.marketing_url:
+        if not latest_revision.external_marketing_url and emeritus_course.marketing_url:
             course_page.external_marketing_url = emeritus_course.marketing_url
             course_page_attrs_changed = True
-        if not course_page.duration and emeritus_course.duration:
+        if not latest_revision.duration and emeritus_course.duration:
             course_page.duration = emeritus_course.duration
             course_page_attrs_changed = True
-        if not course_page.description and emeritus_course.description:
+        if not latest_revision.description and emeritus_course.description:
             course_page.description = emeritus_course.description
             course_page_attrs_changed = True
 
         if course_page_attrs_changed:
-            course_page.save()
+            if course_page.has_unpublished_changes:
+                # If course is in draft, we need to save a revision with the updated data.
+                course_page.save_revision()
+            else:
+                course_page.save()
             log.info(
                 f"Updated external course page for course title: {emeritus_course.course_title}"  # noqa: G004
```

Comment:
I think we can use `save_revision` for both published and draft pages. I've tried to simplify the code which I'm posting here, This works for both published and draft pages:
```python
        latest_revision = course_page.get_latest_revision_as_object()
        # Only update course page fields with API if they are empty in the latest revision.
        # course_page_attrs_changed = False
        if not latest_revision.external_marketing_url and emeritus_course.marketing_url:
            latest_revision.external_marketing_url = emeritus_course.marketing_url
        if not latest_revision.duration and emeritus_course.duration:
            latest_revision.duration = emeritus_course.duration
        if not latest_revision.description and emeritus_course.description:
            latest_revision.description = emeritus_course.description

        latest_revision.save_revision()

        # if course_page_attrs_changed:
        #     if course_page.has_unpublished_changes:
        #         # If course is in draft, we need to save a revision with the updated data.
        #         course_page.save_revision()
        #     else:
        #         course_page.save()
        log.info(
            f"Updated external course page for course title: {emeritus_course.course_title}"  # noqa: G004
        )

```